src/config/ConfigManager.cpp

```diff
@@ -1304,49 +1311,49 @@ void CConfigManager::loadConfigLoadVars() {
 
     // paths
     configPaths.clear();
-
-    std::string              CONFIGPATH;
-
-    static const char* const ENVHOME          = getenv("HOME");
-    const std::string        CONFIGPARENTPATH = ENVHOME + (std::string) "/.config/hypr/";
-
-    if (g_pCompositor->explicitConfigPath == "") {
-        CONFIGPATH = CONFIGPARENTPATH + (ISDEBUG ? "hyprlandd.conf" : "hyprland.conf");
-    } else {
-        CONFIGPATH = g_pCompositor->explicitConfigPath;
+    std::string mainConfigPath = getMainConfigPath();
+    Debug::log(LOG, "Using config: %s", mainConfigPath.c_str());
+    configPaths.push_back(mainConfigPath);
+    std::string configPath = mainConfigPath.substr(0, mainConfigPath.find_last_of('/'));
+    // find_last_of never returns npos since main_config at least has /hypr/
+
+    if (!std::filesystem::is_directory(configPath)) {
+        Debug::log(WARN, "Creating config home directory");
+        try {
+            std::filesystem::create_directories(configPath);
+        } catch (...) {
+            parseError = "Broken config file! (Could not create directory)";
```

Comment:
Why is the config file broken? It's non-existent. This should probably say `Could not create config directory`.

Reply:
This error message existed as-is before this PR, but will be happy to change it.